opensearch-operator/pkg/builders/cluster.go

```diff
@@ -179,12 +179,102 @@ func NewSTSForNodePool(
 	// Supress repeated log messages about a deprecated format for the publish address
 	jvm += " -Dopensearch.transport.cname_in_publish_address=true"
 
-	probe := corev1.Probe{
-		PeriodSeconds:       20,
-		TimeoutSeconds:      5,
-		FailureThreshold:    10,
-		SuccessThreshold:    1,
-		InitialDelaySeconds: 10,
+	startupProbePeriodSeconds := int32(20)
+	startupProbeTimeoutSeconds := int32(5)
+	startupProbeFailureThreshold := int32(10)
+	startupProbeSuccessThreshold := int32(1)
+	startupProbeInitialDelaySeconds := int32(10)
+
+	readinessProbePeriodSeconds := int32(30)
+	readinessProbeTimeoutSeconds := int32(30)
+	readinessProbeFailureThreshold := int32(5)
+	readinessProbeInitialDelaySeconds := int32(60)
+
+	livenessProbePeriodSeconds := int32(20)
+	livenessProbeTimeoutSeconds := int32(5)
+	livenessProbeFailureThreshold := int32(10)
+	livenessProbeSuccessThreshold := int32(1)
+	livenessProbeInitialDelaySeconds := int32(10)
+
+	if node.Probes != nil {
+		if node.Probes.Liveness != nil {
+			if node.Probes.Liveness.InitialDelaySeconds > 0 {
```

Comment:
Are all the following `if` conditions added just to avoid a value 0? Can you also add some defaults in case if user have not passed any values. Also may I know what happens when user passed 0 will it crash without these `if` conditions?

Reply:
If user passes zero then it might fail, i have to check this in k8s if this is the case, I have already added defaults before these if statements.

could you guide me there is better way in golang.

Reply:
I wanted to refrain from adding this kind of conditional statements, but not that adept at golang. If there is a better way with faster processing and lesser code, i would be happy to change this

Reply:
I tired to search a better way, but it seems that this is the only way of retaining the default values here.
This way the user can opt to give one single value change also.